core/agent/agent.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
```

Comment:
This functionality is really service specific, it's core purpose is to generate service config structs.
It's better to add it to the service/common package